pkg/operation/botanist/extensions/dns/dnsowner_test.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dns_test
+
+import (
+	"context"
+	"fmt"
+
+	cr "github.com/gardener/gardener/pkg/chartrenderer"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	mockclient "github.com/gardener/gardener/pkg/mock/controller-runtime/client"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	. "github.com/gardener/gardener/pkg/operation/botanist/extensions/dns"
+	. "github.com/gardener/gardener/test/gomega"
+
+	dnsv1alpha1 "github.com/gardener/external-dns-management/pkg/apis/dns/v1alpha1"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/version"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var _ = Describe("#DNSOwner", func() {
+	const (
+		deployNS     = "test-chart-namespace"
+		dnsOwnerName = "test-deploy"
+		ownerID      = "owner-id"
+	)
+
+	var (
+		ctrl             *gomock.Controller
+		ca               kubernetes.ChartApplier
+		ctx              context.Context
+		c                client.Client
+		expectedDNSOwner *dnsv1alpha1.DNSOwner
+		vals             *OwnerValues
+		defaultDepWaiter component.DeployWaiter
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+
+		ctx = context.TODO()
+
+		s := runtime.NewScheme()
+		Expect(corev1.AddToScheme(s)).NotTo(HaveOccurred())
+		Expect(dnsv1alpha1.AddToScheme(s)).NotTo(HaveOccurred())
+
+		c = fake.NewFakeClientWithScheme(s)
+
+		vals = &OwnerValues{
+			Name:    "test-deploy",
+			Active:  true,
+			OwnerID: ownerID,
+		}
+
+		expectedDNSOwner = &dnsv1alpha1.DNSOwner{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      deployNS + "-" + dnsOwnerName,
+				Namespace: "default",
```

Comment:
Yes, I know. However, the `fake` framework used by this test doesn't seem to support cluster scoped resources. If you omit the namespace, it will be created with a `default` namespace anyway, and subsequent retrieval with an empty namespace would fail. This is why I decided to keep the namespace and set it explicitly to default - I simply couldn't find another way to make this test pass. If you have an idea, let me know.

Reply:
In fact, I found a way to avoid this by properly configuring the RESTMapper used by the test. Check the latest version of this PR.